### ISSUE TITLE
opts_chunk$get("output.dir") contains a mangled file path?

### ISSUE THREAD

**Issue by mtmorgan** [yihui/knitr#1471]
This repo https://github.com/mtmorgan/outputdir has an Rnw vignette. The vignette has a main document and a child document. The code chunk to  include the child document includes fig.path that references `opts_chunk$get("output.dir")`. When built and run in a directory that contains a '.', e.g., 

```
~/tmp/R-3.4.2/check$ R CMD build outputdir && R CMD check outputdir_1.10.0.tar.gz 
```
the check, specifically the step `* checking re-building of vignette outputs ... OK` creates a new directory hierarchy where the '.' in `R-3.4.2` is changed to '_', `R-3_4_2`:

```
$ tree ~/tmp/R-3_4_2/
/home/mtmorgan/tmp/R-3_4_2/
└── check
    └── outputdir_Rcheck
        └── vign_test
            └── outputdir
                └── vignettes
                    ├── Xunnamed-chunk-1-1.pdf
                    └── Xunnamed-chunk-1-1.png

5 directories, 2 files
```
which is very undesirable. I can't tell whether this is an unfortunate name collision, bad form in the original package, or a bug.

**Comment by yihui** [comment]
I'm not sure, either, but this is an undocumented option, and I don't recommend that you use it: https://yihui.name/knitr/options/

**Comment by mtmorgan** [comment]
We'll stop using it, but it seems like its a bug in knitr anyway...

**Comment by yihui** [comment]
Okay, I'll investigate it when I have time. Thanks for the report!

**Comment by yihui** [comment]
@cderv Thanks for the investigation! So the problem is that `output.dir` is an absolute path that contains `.`, and the periods were sanitized by **knitr** due to LaTeX issues. Normally I don't recommend using absolute paths. It seems that @mtmorgan wanted the child Rnw document to generate plots to a specific directory. I don't know what to do here. I can provide an option to disable `knitr:::sanitize_fn()`, if the periods in paths are no longer problems in LaTeX (this was problematic at least around 2013 and I'm not sure if LaTeX is better at dealing with paths containing `.` now).

**Comment by cderv** [comment]
I don't think we have something to do here. `opts_knit$get("output.dir")` is absolute path and it is not to be used in `fig.path` that could be relative. 

This does not seems to be a very common issue, but it is good to have it documented. 

I don't really know about the dot issue in LaTeX.

**Comment by yihui** [comment]
Thanks @mtmorgan, @cderv. As diagnosed in the thread, the mangling came from `knitr:::sanitize_fn()` replacing dots with `_` in figure paths (a historical LaTeX workaround), which also affected absolute `output.dir` paths containing periods.

This was resolved in a007b3762 (#2213): knitr no longer sanitizes dots in figure file paths, and a new option `knitr.sanitize.paths` lets you turn off path sanitization entirely. `knitr:::sanitize_fn("R-3.2.2")` now returns `"R-3.2.2"` unchanged. Closing as fixed.
